src/contracts/custom.ts

```diff
@@ -0,0 +1,61 @@
+import { ContractMetadata } from "../core/classes/contract-metadata";
+import { IStorage, NetworkOrSignerOrProvider } from "../core";
+import { SDKOptions } from "../schema/sdk-options";
+import { ContractWrapper } from "../core/classes/contract-wrapper";
+import {
+  IThirdwebContract,
+  IThirdwebContract__factory,
+} from "@thirdweb-dev/contracts";
+import { CustomContractSchema } from "../schema/contracts/custom";
+import { UpdateableNetwork } from "../core/interfaces/contract";
+
+/**
+ *
+ * @internal
+ */
+export class CustomContract implements UpdateableNetwork {
+  static contractType = "custom" as const;
+  static schema = CustomContractSchema;
+
+  private contractWrapper;
+  public metadata;
+  private storage;
+
+  constructor(
+    network: NetworkOrSignerOrProvider,
+    address: string,
+    storage: IStorage,
+    options: SDKOptions = {},
+    abi: any = undefined,
+    contractWrapper = new ContractWrapper<IThirdwebContract>(
```

Comment:
how about typing the `ContractWrapper` with `BaseContract` instead? that way we don't even require `IThirdwebContract`, should work with any contract and you dont need to do that weird try catch in the sdk.get function?

Reply:
yup fair enough, just means we don't get *anything* on it by default, we need to feature detect every thing (but we already need to do that anyways so I think that's probably fine)